server/app/controllers/ti/TrustedIntermediaryController.java

```diff
@@ -129,17 +131,30 @@ public Result addApplicant(Long id, Http.Request request) {
               + trustedIntermediaryUrl,
           form);
     }
+    return redirectToDashboardWithError(
+        getFormErrors(tiClientCreationResult.getForm().get()),
+        tiClientCreationResult.getForm().get());
+  }
+
+  private String getFormErrors(Form<AddApplicantToTrustedIntermediaryGroupForm> form) {
+    StringBuilder errorMessage = new StringBuilder();
+    form.errors().stream()
+        .forEach(
+            validationError ->
+                errorMessage.append(
+                    System.getProperty("line.separator") + validationError.message()));
+    return errorMessage.toString();
   }
 
   private Result redirectToDashboardWithError(
       String errorMessage, Form<AddApplicantToTrustedIntermediaryGroupForm> form) {
     return redirect(
-            routes.TrustedIntermediaryController.dashboard(Optional.empty(), Optional.empty()))
+            routes.TrustedIntermediaryController.dashboard(Optional.empty(), Optional.of(1)).url())
```

Comment:
suggestion (blocking): please add param comments for the optionals here

Reply:
Sure